### PR TITLE
[autodiff] fix first, second, and higher-order gradients of i0 and i1 at 0.0

### LINKED ISSUE #40627: jax.numpy.i0 drops the second derivative at zero
### Description

`jax.numpy.i0` appears to return an incorrect zero second derivative at zero.

For the modified Bessel function I0:

```
I0(x) = 1 + x**2 / 4 + O(x**4)
```

so:

```
I0''(0) = 1/2
```

In the example below, the differentiated function is `i0(2 * t - 1)`. By the chain rule, the second derivative with respect to `t` at `t = 0.5` should be `2.0`. However, JAX returns `0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import math

os.environ.setdefault("CUDA_VISIBLE_DEVICES", "")
os.environ.setdefault("OMP_NUM_THREADS", "1")
os.environ.setdefault("JAX_PLATFORMS", "cpu")

try:
    import jax
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

jax.config.update("jax_enable_x64", True)

def target(t):
    return jnp.i0(2 * t - 1)

x = jnp.asarray(0.5, dtype=jnp.float64)
expected = 2.0

actual = float(jax.grad(jax.grad(target))(x))

print("actual:", actual)
print("expected:", expected)

if not (math.isfinite(actual) and abs(actual - expected) <= 1e-12):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
actual: 0.0
expected: 2.0
```

### Expected result

```
2.0
```

The forward value is correct, but the second derivative is not. The expected value is finite, normal, and exactly representable in `float64`. JAX returns `0.0`, suggesting that the higher-order autodiff rule for `jax.numpy.i0` loses the curvature at zero.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

### PR DESCRIPTION
**Pull request by jakevdp**
Fixes #40627 and #40628.

Previously, the JVP rule for `jax.numpy.i0` was returning a constant zero for `x=0.0`, which led to incorrect higher-order derivatives. I explored a few solutions, but arrived on using a maclaurin series approach for the JVP of i1, similar to the approach used for `sinc` which has similar autodiff challenges at `x=0.0`.

### CODE REVIEW

**Review by mattjj — APPROVED** [review]
Nice!
